Common/Data/Consolidators/RenkoConsolidator.cs

```diff
@@ -24,53 +24,38 @@ namespace QuantConnect.Data.Consolidators
     /// </summary>
     public class RenkoConsolidator : IDataConsolidator
     {
+        private RenkoBar _currentBar;
+        private DataConsolidatedHandler _dataConsolidatedHandler;
+        private decimal _barSize;
+        private bool _evenBars;
+        private Func<IBaseData, decimal> _selector;
+        private Func<IBaseData, decimal> _volumeSelector;
+
         /// <summary>
-        /// Event handler that fires when a new piece of data is produced
+        /// Gets the kind of the bar
         /// </summary>
-        public event EventHandler<RenkoBar> DataConsolidated;
+        public RenkoType Type => RenkoType.Classic;
 
         /// <summary>
-        /// Event handler that fires when a new piece of data is produced
+        /// Gets a clone of the data being currently consolidated
         /// </summary>
-        event DataConsolidatedHandler IDataConsolidator.DataConsolidated
-        {
-            add { _dataConsolidatedHandler += value; }
-            remove { _dataConsolidatedHandler -= value; }
-        }
-
-        private RenkoBar _currentBar;
-
-        private readonly decimal _barSize;
-        private readonly bool _evenBars;
-        private readonly Func<IBaseData, decimal> _selector;
-        private readonly Func<IBaseData, decimal> _volumeSelector;
-        private DataConsolidatedHandler _dataConsolidatedHandler;
-
-        private bool _firstTick = true;
-        private RenkoBar _lastWicko = null;
+        public IBaseData WorkingData => _currentBar?.Clone();
 
-        private DateTime _openOn;
-        private DateTime _closeOn;
-        private decimal _openRate;
-        private decimal _highRate;
-        private decimal _lowRate;
-        private decimal _closeRate;
+        /// <summary>
+        /// Gets the type consumed by this consolidator
+        /// </summary>
+        public Type InputType => typeof(IBaseData);
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="RenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// Gets <see cref="RenkoBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
         /// </summary>
-        /// <param name="barSize">The constant value size of each bar</param>
-        /// <param name="type">The RenkoType of the bar</param>
-        public RenkoConsolidator(decimal barSize, RenkoType type)
```

Comment:
For a happier user experience suggest we keep this constructor, flag it as obsolete, for example: `[Obsolete("Please use WickedRenkoConsolidator type")]` and make it thrown an exception with a similar message